ts/src/renderer/phaser/classes/PhaserUnit.ts

```diff
@@ -49,6 +50,29 @@ class PhaserUnit extends PhaserAnimatedEntity {
 		this.sprite.setFlip(flip % 2 === 1, flip > 1);
 	}
 
+	protected size (
+		data: {
+			width: number,
+			height: number
+		}
+	): void {
+		super.size(data);
+		if (this.label) {
+			this.updateLabelOffset();
+		}
+		if (this.attributesContainer) {
+			this.updateAttributesOffset();
+		}
+	}
+
+	private updateLabelOffset (): void {
```

Comment:
I think we can combine these into one method and pass a parameter for positive/negative. What do you guys think? Wouldn't take long for me to do but is it necessary?

Reply:
i think it will be less obvious what this method do

Reply:
yeah that's a fair point. it would look like this:
```ts
	private updateOffset (
		child:
			Phaser.GameObjects.GameObject &
			Phaser.GameObjects.Components.Transform,
		direction:
			'above' |
			'below'
	): void {
		child.y =
			(direction === 'above' ? -1 : 1) *
			(25 + (this.sprite.displayHeight + this.sprite.displayWidth) / 4);
	}
	```

Reply:
heard. thanks for the input guys. resolving